mdn/developer-portal
# 1330: Post-redesign HTML, CSS and JS cleanup

This (large...) changeset contains frontend code cleanup following the redesign.

It drops redundant HTML templates, SCSS (files and rules) and JS.

(Related issue #1330)


## How to test

- Code is staged on [the **staging** server](https://developer-portal.stage.mdn.mozit.cloud/) (which is currently acting like a dev server). 
Please compare the _layout_ of pages to [production](https://developer.mozilla.com/), looking for broken flow. 

(I had two tabs open and used space to scroll down equal amounts in each site, doing a blink-comparison between stage and production to confirm there are no layout discrepancies)

Note that images will be different because Production has a set of resized images compared to stage 

----

For the record, here are my notes/local checklists for the removal

```
developerportal/templates/

  [V] means checked
  [D] means deleted
  [/] means commmented

├── [V] 404.html
├── [V] 500.html
├── [V] admin
│   └── [V] login.html
├── [V] atoms - keeping all these because we MAY want to reinstate some later
│   ├── [V] icons
│   │   ├── [V] close.svg
│   │   ├── [V] date.svg
│   │   ├── [V] email.svg
│   │   ├── [V] facebook.svg
│   │   ├── [V] github.svg
│   │   ├── [V] linkedin.svg
│   │   ├── [V] location.svg
│   │   ├── [V] mozilla.svg
│   │   ├── [V] person.svg
│   │   ├── [V] reading-time.svg
│   │   ├── [V] tech-speaker.svg
│   │   ├── [V] twitter.svg
│   │   └── [V] video.svg
│   ├── [V] image.html -> used in rich-text image block
│   ├── [D] img
│   │   ├── [D] person-overlay-1.svg
│   │   ├── [D] person-overlay-2.svg
│   │   └── [D] person-overlay-3.svg
│   ├── [D] pattern.html -> in card-topic.html, article-header.html and pattern-bg-header.html
│   ├── [D] person-icon.html
│   ├── [V] social-meta.html -> used in base.html
│   └── [D] topic-icon.html
├── [V] base.html
├── [V] base_includes
│   └── [V] google_analytics.html
├── [V] footer.html
├── [V] header.html
├── [ ] molecules
│   ├── [D] article-author.html
│   ├── [D] article-details.html
│   ├── [V] card-featured.html
│   ├── [D] card-topic.html
│   ├── [V] cards
│   │   ├── [V] card-article.html
│   │   ├── [V] card-event.html
│   │   ├── [V] card-person.html
│   │   ├── [V] card-video.html
│   │   └── [V] card.html
│   ├── [D] event-details.html
│   ├── [V] filter-form.html
│   ├── [V] header-strip.html
│   ├── [D] icons
│   │   └── [D] topic-default.svg
│   ├── [V] image-block.html
│   ├── [D] item-get-started.html
│   ├── [V] item-topic-link.html
│   ├── [V] navigation
│   │   └── [V] promotion.html
│   ├── [V] pagination.html
│   ├── [D] person-contact.html
│   ├── [V] recent-work-item.html
│   ├── [V] related-links.html
│   ├── [D] resource-meta.html
│   ├── [V] resource-share.html
│   └── [D] resource-topic-tags.html
├── [ ] organisms
│   ├── [D] article-header.html
│   ├── [V] article-read-more.html
│   ├── [/] event-agenda.html
│   ├── [/] event-map.html
│   ├── [/] event-speakers.html
│   ├── [V] featured.html
│   ├── [V] filter-list.html
│   ├── [V] homepage-about.html
│   ├── [V] homepage-header.html
│   ├── [D] latest-cards.html
│   ├── [V] newsletter-signup.html
│   ├── [V] partials
│   │   ├── [D] featured-2-to-5.html
│   │   └── [/] featured-card-selector.html
│   ├── [D] past-events.html
│   ├── [D] pattern-bg-header.html
│   ├── [D] people-section.html
│   ├── [D] social-links.html
│   ├── [V] topic-links.html
│   ├── [V] topic-recent-work.html
│   ├── [D] upcoming-events.html
│   └── [D] video-header.html
├── [V] robots.txt
├── [V] survey.html
└── [V] wagtailadmin
    ├── [V] 404.html
    ├── [V] admin_base.html
    ├── [V] base.html
    ├── [V] home.html
    └── [V] login.html

12 directories, 80 files

- [X] NOW AUDIT JS
  - go through everything in index.js and look for where it's still used, if at all

src/js/
├── [V] admin-extras.js
├── [ ]  atoms
│   ├── [D] map-embed.js + event-map.html
│   ├── [V] modal.js
│   └── [D] toggle.js
├── [V] head-includes.js
├── [V] index.js
├── [V] mozilla-dnthelper.js
├── [V] newsletter-subscription.js
├── [V] organisms
│   ├── [V] filter-form.js
│   └── [V] notification-bar.js
├── [V] polyfills.js
├── [V] task-completion-prompt.js
├── tests
│   ├── [V] newsletter-subscription.test.js
│   └── [V] task-completion-prompt.test.js
└── [V] utils.js

- [ ] NOW AUDIT CSS

Need to look through every file and shed unused classes

[X] means checked
[D] means deleted
[E] means edited

├── [X] admin.scss
├── [X] atoms
│   ├── [X] buttons.scss
│   ├── [X] code.scss
│   ├── [X] embed.scss
│   ├── [X] icon.scss
│   ├── [X] image.scss
│   ├── [X] modal.scss
│   ├── [X] quote.scss
│   ├── [X] rich-text.scss
│   └── [E] typography.scss
├── [X] global
│   ├── [X] base.scss
│   ├── [X] layout.scss
│   └── [X] variables.scss
├── [X] index.scss
├── [X] molecules
│   ├── [X] card-event.scss
│   ├── [X] card-featured.scss
│   ├── [X] card-person.scss
│   ├── [E] card.scss
│   ├── [D] content-hero.scss
│   ├── [X] event-details.scss
│   ├── [X] header-strip.scss
│   ├── [X] header.scss
│   ├── [X] item-topic-link.scss
│   ├── [X] notification-bar.scss
│   ├── [X] pagination.scss
│   ├── [X] resource-share.scss
│   └── [X] resource-toolbar.scss
├── [X] organisms
│   ├── [X] article-read-more.scss
│   ├── [X] event-agenda.scss
│   ├── [X] event-speakers.scss
│   ├── [X] featured.scss
│   ├── [X] filter-form.scss
│   ├── [E] filter-list.scss -- drop card-link and related stuff
│   ├── [X] footer.scss
│   ├── [X] homepage-about.scss
│   ├── [X] homepage-header.scss
│   ├── [X] newsletter-signup.scss
│   ├── [X] people-section.scss
│   └── [X] topic-recent-work.scss
└── [x] templates
    ├── [X] article.scss
    ├── [X] content-page.scss
    ├── [X] event.scss
    ├── [X] events.scss
    ├── [X] http-error.scss
    ├── [X] person.scss
    └── [X] video.scss

- [X] Search for comments in scss files and remove that dead code

Swap in variables not actual vals:
- [X] card.scss -
- [X] footer.scss - swap in variables not actual vals
- [X] homepage-header.scss
- [X] Look for 4/8/12/24/32px hard-coded in CSS

```
